source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -121,119 +121,134 @@ void GeoipProvider::lookup(Geolocation::LookupRequest&& request,
+                            MMDB_CITY_LOOKUP_ARGS[2]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->regionHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->regionHeader().value(),
+                            MMDB_REGION_LOOKUP_ARGS[0], MMDB_REGION_LOOKUP_ARGS[1],
+                            MMDB_REGION_LOOKUP_ARGS[2]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->countryHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->countryHeader().value(),
+                            MMDB_COUNTRY_LOOKUP_ARGS[0], MMDB_COUNTRY_LOOKUP_ARGS[1]);
+  }
+}
+
+void GeoipProvider::populateGeoLookupResultForIspDb(
+    MMDB_lookup_result_s& mmdb_lookup_result,
+    absl::flat_hash_map<std::string, std::string>& lookup_result) const {
+  populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->asnHeader().value(),
+                          MMDB_ASN_LOOKUP_ARGS[0]);
+}
+
+void GeoipProvider::populateGeoLookupResultForAnonDb(
+    MMDB_lookup_result_s& mmdb_lookup_result,
+    absl::flat_hash_map<std::string, std::string>& lookup_result) const {
+  if (config_->isLookupEnabledForHeader(config_->anonHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[0]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->anonVpnHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonVpnHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[1]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->anonHostingHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonHostingHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[2]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->anonTorHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonTorHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[3]);
+  }
+  if (config_->isLookupEnabledForHeader(config_->anonProxyHeader())) {
+    populateGeoLookupResult(mmdb_lookup_result, lookup_result, config_->anonProxyHeader().value(),
+                            MMDB_ANON_LOOKUP_ARGS[4]);
+  }
+}
+
+void GeoipProvider::populateGeoLookupResultForDbType(
+    const std::string& db_type, MMDB_lookup_result_s& mmdb_lookup_result,
+    absl::flat_hash_map<std::string, std::string>& lookup_result) const {
+  if (db_type == "city") {
+    populateGeoLookupResultForCityDb(mmdb_lookup_result, lookup_result);
+  } else if (db_type == "isp") {
+    populateGeoLookupResultForIspDb(mmdb_lookup_result, lookup_result);
+  } else if (db_type == "anon") {
+    populateGeoLookupResultForAnonDb(mmdb_lookup_result, lookup_result);
+  }
+}
+
+void GeoipProvider::lookupInMaxmindDb(
+    const std::string& db_type, const std::string& stat_prefix,
+    const MaxmindDbPtr&
```

Comment:
RFC: `MaxmindDbPtr` is a unique_ptr, so we are assuming the pointer is always available in this function call, would be something to pay attention to when db pointer is reloaded (such as in hot reload)

Reply:
There is no point in taking a const reference to a smart pointer - if you're going to require that it exists for the duration of the call, the function should just take a reference to a `MaxmindDb`.